File: frontend/types/air-quality.ts
export interface Station {
  id: string;
  name_th: string | null;
  name_en: string | null;
  lat: number;
  lon: number;
  province: string | null;
  pm25: number | null;
  aqi: number | null;
  color: string | null;
  level: string | null;
  recorded_at: string | null;
  data_status: "fresh" | "delayed" | "expired";
  age_minutes: number | null;
  eligible_for_surface: boolean;
}

export interface StationsResponse {
  stations: Station[];
  count: number;
  updated_at: string | null;
  fresh_count: number;
  delayed_count: number;
  expired_count: number;
}

export interface HistoryPoint {
  recorded_at: string;
  pm25: number | null;
  aqi: number | null;
}

export interface HistoryResponse {
  station_id: string;
  points: HistoryPoint[];
}

File: backend/models/air_quality.py
"""Air-quality station and historical reading contracts."""

from typing import Literal

from pydantic import BaseModel, Field


class Station(BaseModel):
    id: str
    name_th: str | None = None
    name_en: str | None = None
    lat: float
    lon: float
    province: str | None = None
    pm25: float | None = None
    aqi: int | None = None
    color: str | None = None
    level: str | None = None
    recorded_at: str | None = None
    data_status: str = "expired"
    age_minutes: float | None = None
    eligible_for_surface: bool = False
    in_service_area: bool = False
    quality_flags: list[str] = Field(default_factory=list)


class StationsResponse(BaseModel):
    source: Literal["air4thai"] = "air4thai"
    stations: list[Station]
    count: int
    updated_at: str | None = None
    fresh_count: int = 0
    delayed_count: int = 0
    expired_count: int = 0


class HistoryPoint(BaseModel):
    recorded_at: str
    pm25: float | None = None
    aqi: int | None = None


class HistoryResponse(BaseModel):
    station_id: str
    points: list[HistoryPoint]


class MapHistoryResponse(BaseModel):
    target_at: str
    source: Literal["air4thai"] = "air4thai"
    stations: list[Station]
    count: int
    max_age_minutes: int
